tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/VectorFloat2.java

```diff
@@ -0,0 +1,228 @@
+
+import java.nio.FloatBuffer;
+
+public class VectorFloat2 implements PrimitiveStorage<FloatBuffer> {
+
+    /**
+     * backing array
+     */
+    final protected float[] storage;
+
+    /**
+     * number of elements in the storage
+     */
+    final private int numElements;
+    final private static int elementSize = 2;
+
+    /**
+     * Creates a vector using the provided backing array
+     *
+     * @param numElements
+     * @param array
+     */
+    protected VectorFloat2(int numElements, float[] array) {
+        this.numElements = numElements;
+        this.storage = array;
+    }
+
+    /**
+     * Creates a vector using the provided backing array
+     */
+    public VectorFloat2(float[] array) {
+        this(array.length / elementSize, array);
+    }
+
+    /**
+     * Creates an empty vector with
+     *
+     * @param numElements
+     */
+    public VectorFloat2(int numElements) {
+        this(numElements, new float[numElements * elementSize]);
+    }
+
+    private int toIndex(int index) {
+        return (index * elementSize);
+    }
+
+    /**
+     * Returns the float at the given index of this vector
+     *
+     * @param index
+     *
+     * @return value
+     */
+    public Float2 get(int index) {
+        return loadFromArray(storage, toIndex(index));
+    }
+
+    /**
+     * Sets the float at the given index of this vector
+     *
+     * @param index
+     * @param value
+     */
+    public void set(int index, Float2 value) {
+        value.storeToArray(storage, toIndex(index));
+    }
+
+    /**
+     * Sets the elements of this vector to that of the provided vector
+     *
+     * @param values
+     */
+    public void set(VectorFloat2 values) {
+        for (int i = 0; i < numElements; i++) {
+            set(i, values.get(i));
+        }
+    }
+
+    /**
+     * Sets the elements of this vector to that of the provided array
+     *
+     * @param values
+     */
+    public void set(float[] values) {
+        VectorFloat2 vector = new VectorFloat2(values);
+        for (int i = 0; i < numElements; i++) {
+            set(i, vector.get(i));
+        }
+    }
+
+    public void fill(float value) {
+        for (int i = 0; i < storage.length; i++) {
+            storage[i] = value;
+        }
+    }
+
+    /**
+     * Duplicates this vector
+     *
+     * @return
+     */
+    public VectorFloat2 duplicate() {
+        VectorFloat2 vector = new VectorFloat2(numElements);
+        vector.set(this);
+        return vector;
+    }
+
+    /**
+     * Prints the vector using the specified format string
+     *
+     * @param fmt
+     *
+     * @return
+     */
+    public String toString(String fmt) {
+        String str = "";
+
+        for (int i = 0; i < numElements; i++) {
+            str += get(i).toString() + " ";
+        }
+
+        return str;
+    }
+
+    /**
+     *
+     */
+    public String toString() {
+        if (numElements > 4) {
```

Comment:
same stands for the number 4 that appears in all `toString` methods.

Reply:
this is fixed now